Gems/LyShine/Code/Source/UiCanvasComponent.cpp

```diff
@@ -1874,10 +1874,35 @@ AZ::RHI::AttachmentId UiCanvasComponent::UseRenderTarget(const AZ::Name& renderT
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 AZ::RHI::AttachmentId UiCanvasComponent::UseRenderTargetAsset(const AZ::Data::Asset<AZ::RPI::AttachmentImageAsset>& attachmentImageAsset)
 {
+    // Check that the attachment image asset's bind flags are compatible
+    // - Supports use as a color attachment on a scope
+    // - Supports read access through a ShaderResourceGroup
+    const AZ::RHI::ImageDescriptor& imageDescriptor = attachmentImageAsset->GetImageDescriptor();
```

Comment:
I recommend moving all of this logic to a helper function, probably called 'bool IsAttachmentImageAssetCompatible()' to make readability of the code easier.

Reply:
OK, done.